central/complianceoperator/v2/scanconfigurations/datastore/datastore_impl.go

```diff
@@ -60,23 +60,29 @@ func (ds *datastoreImpl) ScanConfigurationExists(ctx context.Context, scanName s
 
 // GetScanConfigurations retrieves the scan configurations specified by query
 func (ds *datastoreImpl) GetScanConfigurations(ctx context.Context, query *v1.Query) ([]*storage.ComplianceOperatorScanConfigurationV2, error) {
-	if ok, err := scanConfigurationsSAC.ReadAllowed(ctx); err != nil {
-		return nil, err
-	} else if !ok {
-		return nil, nil
-	}
-
-	return ds.storage.GetByQuery(ctx, query)
+	scanConfigs, err := ds.storage.GetByQuery(ctx, query)
+
+	// SAC will return a config if a user has permissions to ANY of the clusters.  For tech preview, and
+	// in the interest of ensuring we don't leak clusters, if a user does not have access to one or more
+	// of the clusters returned by the query, we will return nothing.  An all or nothing approach in the
+	// interest of not leaking data.
+	for _, scanConfig := range scanConfigs {
+		// We must ensure the user has access to all the clusters in a config.  The SAC filter will return the row
+		// if the user has access to any cluster
+		clusterScopeKeys := make([][]sac.ScopeKey, 0, len(scanConfig.GetClusters()))
+		for _, scanCluster := range scanConfig.GetClusters() {
+			clusterScopeKeys = append(clusterScopeKeys, []sac.ScopeKey{sac.ClusterScopeKey(scanCluster.GetClusterId())})
+		}
+		if !complianceSAC.ScopeChecker(ctx, storage.Access_READ_ACCESS).AllAllowed(clusterScopeKeys) {
+			return nil, nil
+		}
+	}
+
+	return scanConfigs, err
 }
 
 // UpsertScanConfiguration adds or updates the scan configuration
 func (ds *datastoreImpl) UpsertScanConfiguration(ctx context.Context, scanConfig *storage.ComplianceOperatorScanConfigurationV2) error {
-	if ok, err := scanConfigurationsSAC.WriteAllowed(ctx); err != nil {
```

Comment:
same comment as above

Reply:
Because of the way SAC works.  Unlike the 'Get', since I added the cluster to the object as a child join table, the SAC filter requires access to all the clusters on the upsert.  I will add a comment.

Reply:
I beefed up `TestUpsertScanConfiguration` in `datastore_impl_test.go` to prove this out.